DataGrid: W1017 and W1018 (#2324)
DevExpress/devextreme-documentation: api-reference/10 UI Components/Errors and Warnings/W1018.md

---
id: ErrorsUIWidgets.W1018
---
---
##### shortDescription

> Occurs in the DataGrid when infinite scrolling is used and **selection**.[allowSelectAll](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/selection/#allowSelectAll) is set to **true**.

Reviewer: I think we should avoid the gerund form ("infinite scrolling"). Do _users_ scroll? If so, can we say the following instead?

`Occurs in the DataGrid when users can scroll infinitely and **selection**.[allowSelectAll](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/selection/#allowSelectAll) is set to **true**.`

Author: The gerund form can be here because there's the [scrolling ](https://js.devexpress.com/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/scrolling/) prop.

---

> To select all rows, the component requires their total number. With infinite scrolling, the component cannot calculate it because rows are loaded dynamically.

Reviewer: 1) `To select all rows, the component requires their total number.`

What do you mean by "their"? Please specify.

2) `With infinite scrolling, the component cannot calculate it because rows are loaded dynamically.`

We should avoid the gerund, and specify _who_ scrolls. Is it the user?

Also, what do you mean by "calculate it"? What are you calculating?

Author: "their total number" means the total number of rows.
"the component cannot calculate" means that the component can't count rows to get the total number of rows.

> We should avoid the gerund, and specify who scrolls. Is it the user?
Replied in the comment for line #6.

Reviewer 2: ```suggestion
To select all rows, the component requires their total number. The component cannot calculate the total number when infinite scrolling is used because rows are loaded dynamically.
```

> You can do one of the following to get rid of this warning:

Reviewer: Since this section is called "Errors and Warnings", instead of "to get rid of this warning" can we say "to fix this error"?

Author: "To fix this, you can do one of the following:" is better. What about that?

 
- **Use virtual scrolling instead**            

> With [virtual scrolling](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/scrolling/#mode), the component knows the total number of rows and can select all rows. [View Demo](https://js.devexpress.com/Demos/WidgetsGallery/Demo/DataGrid/VirtualScrolling/)

Reviewer: ```suggestion
If the [virtual scroll](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/scrolling/#mode) feature is activated, the component can calculate the total number of rows and select all rows. [View Demo](https://js.devexpress.com/Demos/WidgetsGallery/Demo/DataGrid/VirtualScrolling/)
```
